src/kernel-install/90-loaderentry.install

```diff
@@ -21,51 +21,51 @@ MACHINE_ID=$KERNEL_INSTALL_MACHINE_ID
 BOOT_DIR="/$MACHINE_ID/$KERNEL_VERSION"
 BOOT_ROOT=${BOOT_DIR_ABS%$BOOT_DIR}
 
-if [[ $COMMAND == remove ]]; then
+if [ "$COMMAND" = remove ]; then
     rm -f "$BOOT_ROOT/loader/entries/$MACHINE_ID-$KERNEL_VERSION.conf"
     rm -f "$BOOT_ROOT/loader/entries/$MACHINE_ID-$KERNEL_VERSION+"*".conf"
     exit 0
 fi
 
-if ! [[ $COMMAND == add ]]; then
+if ! [ "$COMMAND" = add ]; then
     exit 1
 fi
 
-if ! [[ $KERNEL_IMAGE ]]; then
+if [ -z "$KERNEL_IMAGE" ]; then
     exit 1
 fi
 
-if [[ -f /etc/os-release ]]; then
+if [ -f /etc/os-release ]; then
     . /etc/os-release
-elif [[ -f /usr/lib/os-release ]]; then
+elif [ -f /usr/lib/os-release ]; then
     . /usr/lib/os-release
 fi
 
-if ! [[ $PRETTY_NAME ]]; then
+if [ -z "$PRETTY_NAME" ]; then
     PRETTY_NAME="Linux $KERNEL_VERSION"
 fi
 
-declare -a BOOT_OPTIONS
-
-if [[ -f /etc/kernel/cmdline ]]; then
-    read -r -d '' -a BOOT_OPTIONS < /etc/kernel/cmdline
+if [ -f /etc/kernel/cmdline ]; then
+    read -r BOOT_OPTIONS < /etc/kernel/cmdline
 fi
 
-if ! [[ ${BOOT_OPTIONS[*]} ]]; then
-    read -r -d '' -a line < /proc/cmdline
-    for i in "${line[@]}"; do
-        [[ "${i#initrd=*}" != "$i" ]] && continue
-        BOOT_OPTIONS+=("$i")
-    done
+if [ -z "${BOOT_OPTIONS}" -a -f "/proc/cmdline" ]; then
+    BOOT_OPTIONS=$(
+        cat /proc/cmdline | tr ' ' '\n' | \
```

Comment:
http://porkmail.org/era/unix/award.html#cat :)

Reply:
Yeah… though `tr ' ' '\n' < /proc/cmdline | …` looks a bit weird tbh.

Reply:
Wouldn’t this do the same?
```sh
for i in $(cat /proc/cmdline)
do
  [ "${i#initrd=*}" != "$i" ] || echo -n " $i"
done
```

Reply:
Or even (in this context):
```sh
BOOT_OPTIONS="$(for i in $(cat /proc/cmdline)
do
  [ "${i#initrd=*}" != "$i" ] || echo $i
done)"
```